azerothcore/azerothcore-wotlk
# feat(Command): Premium accounts and characters

## CHANGES PROPOSED:
-  Added tables on database to support premium account and characters
-  Added command script to insert, delete and read the tables

Barbz requested a general solution to many module problems, allowing scaling the potential of costumizable content in Azerothcore.
These tables will allow to define premium / VIP / special account and characters for a server. This way modules are given to possibility to use these tables as they see fit to their needs. There was the need of Azerothcore to support fellow moders and I propose this solution.
So far I've implemented these commands but more can be requested eventually as they are needed.

## TESTS PERFORMED:
- Built Windows 10
- Tested in-game commands:

### Command's list
1. premium character info
2. premium character create
3. premium character delete
4. premium account info
5. premium account create
6. premium account delete




## HOW TO TEST THE CHANGES:

1. premium info of both account and character should give the premium level for each one. If there isn't a premium level to any, an error message will be delivered.
2. premium create inserts on the database for characters and accounts acordingly. Auth stores premium accounts and characters stores premium characters. If you try to override a character with a new premium level, the gamemaster will be asked to delete first, then insert. The same will happen for account.
3. premium delete will, as the command so specifies, remove the premium level for said character or account. If there's no level set, a message will clarify it.

Note: To set an account premium only needs to target a player and the script will look for that player's account so the only parameters that we need to send to the console are the premium levels when creating.


## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
